src/host/screenInfo.cpp

```diff
@@ -2277,6 +2277,55 @@ void SCREEN_INFORMATION::SetViewport(const Viewport& newViewport,
     return S_OK;
 }
 
+// Method Description:
+// - Clear the entire contents of the viewport, except for the cursor's row,
+//   which is moved to the top line of the viewport.
+// - This is used exclusively by ConPTY to support GH#1193, GH#1882. This allows
+//   a terminal to clear the contents of the ConPTY buffer, which is important
+//   if the user would like to be able to clear the terminal-side buffer.
+// Arguments:
+// - <none>
+// Return Value:
+// - S_OK
+[[nodiscard]] HRESULT SCREEN_INFORMATION::ClearBuffer()
+{
+    const COORD oldCursorPos = _textBuffer->GetCursor().GetPosition();
+    short sNewTop = oldCursorPos.Y;
+    const Viewport oldViewport = _viewport;
+
+    short delta = (sNewTop + _viewport.Height()) - (GetBufferSize().Height());
+    for (auto i = 0; i < delta; i++)
+    {
+        _textBuffer->IncrementCircularBuffer();
+        sNewTop--;
+    }
+
+    const COORD coordNewOrigin = { 0, sNewTop };
+    RETURN_IF_FAILED(SetViewportOrigin(true, coordNewOrigin, true));
+
+    // Place the cursor at the same x coord, on the row that's now the top
+    RETURN_IF_FAILED(SetCursorPosition(COORD{ oldCursorPos.X, sNewTop }, false));
+
+    // Update all the rows in the current viewport with the standard erase attributes,
+    // i.e. the current background color, but with no meta attributes set.
+    auto fillAttributes = GetAttributes();
+    fillAttributes.SetStandardErase();
+
+    // +1 on the y coord because we don't want to clear the attributes of the
+    // cursor row, the one we saved.
+    auto fillPosition = COORD{ 0, _viewport.Top() + 1 };
+    auto fillLength = gsl::narrow_cast<size_t>(_viewport.Height() * GetBufferSize().Width());
+    auto fillData = OutputCellIterator{ fillAttributes, fillLength };

```

Comment:
```suggestion
    const auto fillPosition = COORD{ 0, _viewport.Top() + 1 };
    const auto fillLength = gsl::narrow_cast<size_t>(_viewport.Height() * GetBufferSize().Width());
    const auto fillData = OutputCellIterator{ fillAttributes, fillLength };
```
Not sure if these can be const, but if they can, there ya go.